### PR TITLE
chore: remove axios

### PR DESCRIPTION
**Pull request by enpitsuLin**
### <samp>🤖 Generated by Copilot at ad2d70e</samp>

This pull request removes the `axios` library from the project and replaces it with the native fetch API for making HTTP requests. This reduces the project dependencies, simplifies the code, and improves the compatibility with IPFS. The pull request affects the `pnpm-lock.yaml` file and several files in the `src` folder.

### <samp>🤖 Generated by Copilot at ad2d70e</samp>

> _Oh we're the brave developers of the `langchain` crew_
> _We don't need no `axios` to make our requests go through_
> _We use the native `fetch` API and simplify our code_
> _And we check for nulls and errors as we sail the IPFS node_

### WHY

axios is not very useful in modern network technologies, and its size is also quite large. replacing it with fetch will reduce the bundler size.

### <samp>🤖 Generated by Copilot at ad2d70e</samp>

*  Remove axios dependency and references from the project ([link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL103-L105), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL165-R162), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9596-R9593), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9691), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-f4d932c60075f2ab208b12cf771a3ff68a1f22688e7e03c7dbcd65c82ca49792L1), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-33703f17aa564b3fcd73bc6e2105a297370db462e5dc992c284e337905e3fed0L1))
*  Replace axios with native fetch API for making HTTP requests in `UniMedia.tsx`, `upload-file.ts`, and `page.model.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-f4d932c60075f2ab208b12cf771a3ff68a1f22688e7e03c7dbcd65c82ca49792L19-R24), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-8e5b7f1ac7499dd1ce97f0cabf0b05c83bdea2a41fd891cbaa7da263967c5978L1-R10), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-33703f17aa564b3fcd73bc6e2105a297370db462e5dc992c284e337905e3fed0L348-R354))
*  Handle null or undefined cases for content-type header and input parameters using nullish coalescing operator and early return in `UniMedia.tsx` and `page.model.ts` ([link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-f4d932c60075f2ab208b12cf771a3ff68a1f22688e7e03c7dbcd65c82ca49792L19-R24), [link](https://github.com/Crossbell-Box/xLog/pull/469/files?diff=unified&w=0#diff-33703f17aa564b3fcd73bc6e2105a297370db462e5dc992c284e337905e3fed0L348-R354))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

**Comment by Innei** [comment]
@DIYgod What is the minimum required Node.js version that needs to be supported currently?

File: src/models/page.model.ts [hunk 1]
```diff
@@ -345,14 +344,14 @@ export async function getPage<TRender extends boolean = false>(
...
-        })
-      ).data
+      if (!input.site || !input.page) return null
+      const response = await fetch(
+        `/api/slug2id?${new URLSearchParams({
+          handle: input.site,
+          slug: input.page,
+        }).toString()}`,
+      )
+      const slug2Id = (await response.json()).data
```
Comment (DIYgod) [inline src/models/page.model.ts:354]: `.data` here should be removed

**Review by DIYgod — CHANGES_REQUESTED** [review]

**Comment by DIYgod** [comment]
@Innei The minimum required version of Node.js now is 15, but we can change it to 18.

**Comment by Songkeys** [comment]
Next.js has the fetch polyfill baked in since 9.4

**Pull request merged**